lib/types.nix

```diff
@@ -93,4 +93,39 @@ rec {
         optionDescriptionPhrase (class: class == "noun" || class == "composite") elemType
       }";
     };
+
+  pluginLuaConfig = types.submodule (
+    { config, ... }:
+    {
+      options = {
+        pre = mkOption {
+          type = types.lines;
+          default = "";
+          description = "Configuration before the initialization of the plugin";
+        };
+        post = mkOption {
+          type = types.lines;
+          default = "";
+          description = "Configuration after the initialization of the plugin";
+        };
+        init = mkOption {
+          type = types.lines;
+          internal = true;
+          default = "";
+          description = "Initialization code (between pre & post)";
+        };
+        final = mkOption {
+          type = types.str;
+          description = "Complete plugin configuration";
+          readOnly = true;
+        };
```

Comment:
Should we consider marking this as `internal`?

I'm also wondering if it makes sense to make it a `types.lines` and not `readOnly`?

We could then define the value as:
```nix
config.final = mkMerge [
  (mkIf (config.pre != "") (mkBefore config.pre))
  (mkIf (config.init != "") config.init)
  (mkIf (config.post != "") ((mkAfter config.post))
]
```

Reply:
I think it makes sense to mark it internal, since it's going to be a derived value from other options. I'm not sure on benefit of readOnly vs not.

Reply:
Personally I see `readOnly` as a "last resort" for when it absolutely never makes sense for users to modify the option's value.

I think `internal` makes more sense here, and if the option is mutable I think `types.lines` makes more sense than `types.str`.

Reply:
The idea of it being internal is that you would be able to benefit from nixvim genrrating the setup code, but you could put it anywhere, in an autocmd for example.

In that case I would strongly prefer this to be readOnly

Reply:
> The idea of it being internal 

Currently it isn't `internal`. I'm suggesting maybe it should be?

> you would be able to benefit from nixvim genrrating the setup code, but you could put it anywhere, in an autocmd for example

I don't see how it being mutable is an obstacle to any of that?

I guess we could advise users that they should normally prefer the `pre` and `post` options, but aren't we just imposing needless restrictions by making `final` read-only?

Reply:
Yeah I was detailing my reasons for it _not_ to be internal.

It is indeed a bit of a "useless" restriction, but I don't like very much to have two ways to do something (add to pre/post or add to final).
I think pre/post is better for newcommers, so that's why I went with this approach

Reply:
I think internal would make it hidden from a newcomer, though. It just gives power to advanced users who are aware they might shoot themselves in the foot by using an option. But, I'm not opposed to just revisiting later if we find a desire to override the final option, at some point.

Reply:
Yeah I don't want it to be internal, though my first reaction was to mark it as readOnly to be conservative in what we expose. If we judge there is a sufficient value in marking it not readOnly I could do it.

I'm not a 100% convinced it's needed with the pre/post options though